blog/2024-05-01-improved-hass-data-typing.md

```diff
@@ -0,0 +1,68 @@
+---
+author: Marc Mueller
+authorURL: https://github.com/cdce8p
+title: "Improved typing for hass.data"
+---
+
+In the past, one of the challenges with `hass.data` was to correctly assign type information. Since it was typed as `dict[str, Any]`, the only options were annotation assignments or `cast` like:
+
+```py
+data: MyData = hass.data[SOME_KEY]
+```
+
+This had several disadvantages. Not only was it necessary to annotate ever assignment, type checkers basically pretended that the annotation would always be correct. Especially during refactoring, it could easily happen that one instance was missed and, while type-checking still succeeded, the actual code would be broken.
```

Comment:
```suggestion
This had several disadvantages. Not only was it necessary to annotate every assignment, but type checkers also basically pretended that the annotation would always be correct. Especially during refactoring, it could easily happen that one instance was missed, and while type-checking still succeeded, the actual code would be broken.
```